forge-gui-desktop/src/main/java/forge/screens/home/VLobby.java

```diff
@@ -245,29 +223,34 @@ public void focusOnAvatar() {
     }
 
     @Override
-    public void update(final int slot, final LobbySlotType type){
+    public void update(final int slot, final LobbySlotType type) {
         final FDeckChooser deckChooser = getDeckChooser(slot);
-        deckChooser.setIsAi(type==LobbySlotType.AI);
-        DeckType selectedDeckType = deckChooser.getSelectedDeckType();
-        switch (selectedDeckType){
-            case STANDARD_CARDGEN_DECK:
-            case PIONEER_CARDGEN_DECK:
-            case HISTORIC_CARDGEN_DECK:
-            case MODERN_CARDGEN_DECK:
-            case LEGACY_CARDGEN_DECK:
-            case VINTAGE_CARDGEN_DECK:
-            case COLOR_DECK:
-            case STANDARD_COLOR_DECK:
-            case MODERN_COLOR_DECK:
-                deckChooser.refreshDeckListForAI();
-                break;
-            default:
-                break;
+        final boolean ai = type == LobbySlotType.AI;
+        // Only change decks when switching to AI, but not when, for example switching AI types.
+        if (ai && !deckChooser.isAi()) {
```

Comment:
Just want to mention I saw this reply and am planning to investigate a bit more.

Reply:
So to give more details about what's happening...

First, I'm testing this with Tiny Leaders variant.

How the code works, is each format has its own FDeckChooser (with lots of code duplication). The Tiny Leaders deck chooser is initialized like so:
```
            final FDeckChooser fdtlcom = view.getTinyLeaderDeckChooser(iSlot);
            fdtlcom.initialize(FPref.TINY_LEADER_DECK_STATES[iSlot], defaultDeckTypeForTinyLeaderSlot(iSlot));
            fdtlcom.populate();
```
```
    private static DeckType defaultDeckTypeForTinyLeaderSlot(final int iSlot) {
        return iSlot == 0 ? DeckType.TINY_LEADERS_DECK : DeckType.RANDOM_CARDGEN_COMMANDER_DECK;
    }
```

So for P2, the type is DeckType.RANDOM_CARDGEN_COMMANDER_DECK. This doesn't seem to get updated at all later, even though when the saved P2 deck comes from "Custom User Decks" and when that's actually what's shown in the UI. So when you toggle AI player type (or just click the AI radio button even when it's selected), it tries to pick a deck from RANDOM_CARDGEN_COMMANDER_DECK list, resetting whatever deck was selected.

So, theoretically, the fix should be to set the decktype properly (when loading prefs) on the P2 tiny leaders deck list.

Reply:
Debugging more, it seems the non-TL commander deck chooser gets an update via this code, which results in an incorrect update of the TL selected deck. The actual call that causes the update is this:

```
         updateCommanderStyleDeckChooser(getCommanderDeckChooser(slot), type);
```

As opposed to:
```
        updateCommanderStyleDeckChooser(getTinyLeaderDeckChooser(slot), type);
``` 

Now, the question is why. I find this code is unfortunately quite messy because it's keeping a lot of state around unnecessarily - for example, all the UI components for all the variants. Ideally, it would just create the UI components it needs based on the selected settings - any other state would just be persisted in prefs. I think that would simplify things a lot and avoid these types of bugs. I'll take a look if that can be done.

Reply:
Sent https://github.com/Card-Forge/forge/pull/2139 instead that fixes this via a refactoring & clean up.